porch/pkg/cache/repository.go

```diff
@@ -68,6 +72,20 @@ func (c *cachedPackageRevision) GetPackageRevision() *v1alpha1.PackageRevision {
 
 var _ repository.PackageRevision = &cachedPackageRevision{}
 
+type cachedPackage struct {
+	repository.Package
+	latestPackageRevision string
+}
+
+var _ repository.Package = &cachedPackage{}
+
+func (c *cachedPackage) GetLatestRevision() string {
+	if c.latestPackageRevision != "" {
+		return c.latestPackageRevision
+	}
+	return c.Package.GetLatestRevision()
```

Comment:
I think this will be a nontrivial operation, though I haven't fully thought out the implementation yet. I think the implementation will be something like getting all the package revisions related to this package (possibly through git, or or possibly from the already cached revisions), and then calculating the latest revision for each package with semver comparisons. 

It's not necessary to cache if we are okay with iterating over the package revisions each time, but that seems like a lot of duplicate work to me.